docs/advanced-topics/hybrid-models.md

```diff
@@ -0,0 +1,136 @@
+
+However, not all applications can be easily converted to FHE computation and the computation cost of FHE may make a full conversion exceed latency requirements.
+
+Hybrid models provide a balance between on-device deployment and cloud-based deployment. This approach entails executing parts of the model directly on the client side, while other parts are securely processed with FHE on the server side. Concrete ML facilitates the hybrid deployment of various neural network models, including MLP (multilayer perceptron), CNN (convolutional neural network), and Large Language Models.
+
+{% hint style="warning" %}
+If model IP protection is important, care must be taken in choosing the parts of a model to be executed on the cloud. Some
+black-box model stealing attacks rely on knowledge distillation
+or on differential methods. As a general rule, the difficulty
+to steal a machine learning model is proportional to the size of the model, in terms of numbers of parameters and model depth.
+{% endhint %}
+
+The hybrid model deployment API provides an easy way to integrate the [standard deployment procedure](client_server.md) into neural network style models that are compiled with [`compile_brevitas_qat_model`](../developer-guide/api/concrete.ml.torch.compile.md#function-compile_brevitas_qat_model) or [`compile_torch_model`](../developer-guide/api/concrete.ml.torch.compile.md#function-compile_torch_model).
+
+## Compilation
+
+To use hybrid model deployment, the first step is to define what part of the PyTorch neural network model must be executed in FHE. The model part must be a `nn.Module` and is identified by its key in the original model's `.named_modules()`.
+
+```python
+import numpy as np
+import os
+import torch
+
+from pathlib import Path
+from torch import nn
+
+from concrete.ml.torch.hybrid_model import HybridFHEModel, tuple_to_underscore_str
+from concrete.ml.deployment import FHEModelServer
+
+
+class FCSmall(nn.Module):
+    """Torch model for the tests."""
+
+    def __init__(self, dim):
+        super().__init__()
+        self.seq = nn.Sequential(nn.Linear(dim, dim), nn.ReLU(), nn.Linear(dim, dim))
+
+    def forward(self, x):
+        return self.seq(x)
+
+model = FCSmall(10)
+model_name = "FCSmall"
+submodule_name = "seq.0"
+
+inputs = torch.Tensor(np.random.uniform(size=(10, 10)))
+# Prints ['', 'seq', 'seq.0', 'seq.1', 'seq.2']
+print([k for (k, _) in model.named_modules()])
+
+# Create a hybrid model
+hybrid_model = HybridFHEModel(model, [submodule_name])
+hybrid_model.compile_model(
+    inputs,
+    n_bits=8,
+)
+
+
+models_dir = Path(os.path.abspath('')) / "compiled_models"
```

Comment:
`Path(os.path.abspath(''))` is identical to `Path('.')>resolve()` right ? (avoid importing `os`)

Reply:
I like `os.path` :)